File: day-02/py/part2.py
with open("input.txt", "r") as file:
    safe_count = 0
    for line in file:
        reports = [int(x) for x in line.strip().split()]
        if len(reports) > 1:
            for toRemove in range(len(reports)):
                safe = True
                dupReports = reports[:toRemove] + reports[toRemove + 1:]
                order = 1 if dupReports[1] > dupReports[0] else -1
                for i in range(1, len(dupReports)):
                    diff = dupReports[i] - dupReports[i - 1]
                    if order == 1:
                        if not (1 <= diff <= 3):
                            safe = False
                            break
                    else:
                        if not (-3 <= diff <= -1):
                            safe = False
                            break
                if safe:
                    safe_count += 1
                    break

print(safe_count)

File: day-02/c/part2.c
#include <stdio.h>
#include <stdbool.h>
#include "../../utils/io_utils.h"

#define M 100

int main(void) {
    int reports[M], dupReports[M - 1], n, order, i, j, i2, diff, safe_count = 0;
    bool safe;
    while ((n = read_ints(stdin, reports))) {
        for (j = 0; j < n; j++) {
            safe = true;
            for (i2 = 0, i = 0; i2 < n; i2++) {
                if (i2 == j) continue;
                dupReports[i++] = reports[i2];
            }
            if (n - 1 > 1) {
                order = dupReports[1] > dupReports[0] ? 1 : -1;
                for (i = 1; i < n - 1; i++) {
                    diff = dupReports[i] - dupReports[i - 1];
                    if (order == 1) {
                        if (!(diff >= 1 && diff <= 3)) {
                            safe = false;
                            break;
                        }
                    } else {
                        if (!(diff >= -3 && diff <= -1)) {
                            safe = false;
                            break;
                        }
                    }
                }
            }
            if (safe) {
                safe_count++;
                break;
            }
        }
    }
    printf("%d\n", safe_count);
    return 0;
}